modules/mqtt/destination/mqtt-destination.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2021 One Identity
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "mqtt-destination.h"
+#include "mqtt-worker.h"
+#include "mqtt-parser.h"
+
+#include "plugin.h"
+#include "messages.h"
+#include "misc.h"
+#include "stats/stats-registry.h"
+#include "logqueue.h"
+#include "driver.h"
+#include "plugin-types.h"
+#include "logthrdest/logthrdestdrv.h"
+
+/*
+ * Default values
+ */
+
+#define DEFAULT_HOST "localhost"
+#define DEFAULT_PORT 1833
+#define DEFAULT_CLEAN_SESSION FALSE
+#define DEFAULT_KEEPALIVE 60
+#define DEFAULT_QOS 0
+#define DEFAULT_PROTOCOL_TYPE "tcp"
+#define DEFAULT_MESSAGE_TEMPLATE "$MESSAGE"
+
+
+/*
+ * Configuration
+ */
+
+void
+mqtt_destination_dd_set_host (LogDriver *d, const gchar *host)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  g_string_assign(self->host, host);
+}
+
+void
+mqtt_destination_dd_set_port (LogDriver *d, const gint port)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  self->port = port;
+}
+
+void
+mqtt_destination_dd_set_topic(LogDriver *d, const gchar *topic)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  g_string_assign(self->topic, topic);
+}
+
+
+void
+mqtt_destination_dd_set_keepalive (LogDriver *d, const gint keepalive)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  self->keepalive = keepalive;
+}
+
+void
+mqtt_destination_dd_set_qos (LogDriver *d, const gint qos)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  self->qos = qos;
+}
+
+void
+mqtt_destination_dd_set_protocol_type(LogDriver *d, const gchar *protocol_type)
+{
+  MQTTDestinationDriver *self = (MQTTDestinationDriver *)d;
+  g_string_assign(self->protocol_type, protocol_type);
+}
+
+void
+mqtt_destination_dd_set_message_ref(LogDriver *d, LogTemplate *message)
```

Comment:
I would indicate that this is a template in the function name, like using set_message_template_ref()

Reply:
solved by this commit: rename message template setter
(commit hash: ad74e5cd4b38c8ecd4767ef0ea65c5da97fd752d )